crates/noirc_evaluator/src/ssa/code_gen.rs

```diff
@@ -399,7 +399,10 @@ impl<'a> IRGenerator<'a> {
     /// This function could use a clearer name
     fn bind_fresh_pattern(&mut self, basename: &str, typ: &Type, value: Value) -> Value {
         match value {
-            Value::Single(node_id) => self.bind_variable(basename.to_owned(), None, typ, node_id),
+            Value::Single(node_id) => {
+                let otype = self.context.get_object_type(node_id);
```

Comment:
`typ.into()` was failing with arrays, and since we already have the type, we use it directly.

Reply:
Is `typ.into()` failing because it gets an array's element type rather than the type itself, or was this behavior changed? I'm thinking we should change that behavior or rename the function since it is confusing. I would expect `typ.into() == self.context.get_object_type(node_id)` to hold.

Reply:
yes it is because of the array and no the behavior did not change. The get_object_type and typ.into() are not referring to the same object, this is why the proper fix is to use the already existing type.